crates/swc_ecma_transforms_base/src/resolver/mod.rs

```diff
@@ -549,6 +549,18 @@ impl<'a> VisitMut for Resolver<'a> {
 
             let old = child.ident_type;
             child.ident_type = IdentType::Binding;
+            {
+                let params = e
+                    .params
+                    .iter()
+                    .filter(|p| !p.is_rest())
+                    .flat_map(find_pat_ids)
+                    .collect::<Vec<Id>>();
```

Comment:
Do we need to collect?

Reply:
We are iterating right after collecting, so `collect` may not worth

Reply:
That's how this PR works -- you may refer latter declared parameters in closures, so we must hoist all parameters before we visit them.

Reply:
Same as L893 in `visit_mut_function`

Reply:
I see, thank you!